src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -58,33 +58,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     String userId = getStringParameter(request, USER_ID_PROPERTY, DEFAULT_STRING);
     PreparedQuery results = datastore.prepare(new Query(userId));
```

Comment:
Something I think I suggested in the test that I think makes sense here. You would change the `if (count == 0)` check to `if (entity is null)`:

```
Entity entity = datastore.prepare(new Query(userId)).asSingleEntity();
```